### DISCUSSION TITLE
How to get code coverage with c8 rather than istanbul

### DISCUSSION THREAD

**Discussion by Ancient-Dragon** [cypress-io/cypress#26485] [Questions and Help]
Hi there, we're facing issues with merging coverage from our vitest and cypress tests. Looking at the issue a lot of people are suggesting moving to c8 instead of using istanbul / nyc. I don't see a way for us to do this in cypress but I might be missing something.

**Comment by portyu9** [comment]
Hi @Ancient-Dragon 

There is no supported switch in `@cypress/code-coverage` that makes Cypress collect coverage through c8/V8 instead of Istanbul.

The Cypress coverage plugin expects the application to be instrumented for Istanbul coverage and reads the resulting `window.__coverage__` data.

If your goal is to combine Vitest and Cypress coverage, the simplest common format is therefore to configure Vitest to use its `istanbul` provider too:

```ts
export default defineConfig({
  test: {
    coverage: {
      provider: 'istanbul',
    },
  },
})
```

Then instrument the application exercised by Cypress with Istanbul and merge the resulting Istanbul coverage data/reports.

Modern Vitest also supports its `v8` provider, but that does not make `@cypress/code-coverage` collect native V8/c8 coverage. If you keep Vitest on V8, you need a separate conversion/report-merging pipeline outside of the Cypress plugin.

So if interoperability between the two test suites is the main goal, using Istanbul on both sides is currently the simplest supported approach.

Please mark this answer as accepted if it helped, thank you.
